sycl/source/detail/jit_compiler.cpp

```diff
@@ -694,6 +705,25 @@ jit_compiler::fuseKernels(QueueImplPtr Queue,
       return A.MIndex < B.MIndex;
     });
 
+    // Determine whether the kernel has been subject to DPCPP's range rounding.
+    // If so, the first argument will be the original ("user") range.
+    std::optional<size_t> UserGlobalSize;
+    if ((KernelName.find("_ZTSN4sycl3_V16detail18RoundedRangeKernel") == 0 ||
+         KernelName.find("_ZTSN4sycl3_V16detail19__pf_kernel_wrapper") == 0) &&
+        !Args.empty()) {
+      auto &A0 = Args[0];
+      auto Dims = KernelCG->MNDRDesc.Dims;
+      if (A0.MPtr && A0.MSize == static_cast<int>(Dims * sizeof(size_t)) &&
+          A0.MType == kernel_param_kind_t::kind_std_layout) {
```

Comment:
Is there any chance we get here and we do not enter this `if` statement? Would that mean range rounding is performed and there is no first argument user range? Should we carry on fusion in that case or we could run into illegal fusion?

Reply:
Right, no, this is a compiler error (something changed in the RR transformation that we're not aware of). The check is now an `assert`.